Add newsletter entry for new multiapps features

GiudGiud opened the pull request:
refs https://github.com/idaholab/moose/pull/11447

GiudGiud committed:
Add newsletter entry for new multiapps features, refs #11447

cticenhour reviewed (changes requested):
Might revise this after I see the render, but here's what I saw just looking at the code.

GiudGiud committed:
Apply suggestions from code review

Co-authored-by: Casey Icenhour <cticenho@ncsu.edu>

Apply suggestions from code review

Co-authored-by: Casey Icenhour <cticenho@ncsu.edu>

Apply suggestions from code review

Co-authored-by: Casey Icenhour <cticenho@ncsu.edu>

Update modules/doc/content/newsletter/2023/2023_02.md

Co-authored-by: Casey Icenhour <cticenho@ncsu.edu>

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1348181/) on 8126938 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/23421/site)

This comment will be updated on new commits.

cticenhour commented on modules/doc/content/newsletter/2023/2023_02.md:
```diff
@@ -7,8 +7,39 @@ for a complete description of all MOOSE changes.
+### New MultiApp general field transfers
+
+General field transfers have been added to the framework. They are a re-implementation of existing transfers with additional functionalities,
+based on libMesh's [generic projector](https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1GenericProjector.html). The [documentation for the
+base class](MultiAppGeneralFieldTransfer.md) explains all the features (siblings transfers, spatial restrictions, higher order variables, etc.) which are supported. The specific instances implemented are:
```
```suggestion
based on libMesh's [generic projector](https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1GenericProjector.html). The [documentation for the base class](MultiAppGeneralFieldTransfer.md)
explains all the features (siblings transfers, spatial restrictions, higher order variables, etc.) which are supported. The specific instances implemented are:
```

Gotta keep the link text on the same line or it breaks.

cticenhour reviewed (changes requested):
Render looked good other than this. Should be ready to merge after this is fixed.
